src/css-select-parse5-adapter.ts

```diff
@@ -0,0 +1,94 @@
+import {Adapter as CSSSelectAdapter} from 'css-select';
+import {Attribute, Element, Node, TreeAdapter} from 'parse5';
+const defaultTreeAdapter = require('parse5/lib/tree-adapters/default');
+export type Predicate = (node: Node) => boolean;
+export class Adapter implements CSSSelectAdapter<Node, Element> {
+  treeAdapter: TreeAdapter;
+  constructor(treeAdapter: TreeAdapter) {
+    this.treeAdapter = treeAdapter;
+  }
+  isTag(node: Node): node is Element {
+    return this.treeAdapter.isElementNode(node);
+  }
+  existsOne(test: Predicate, elems: Element[]): boolean {
+    return elems.some(test);
+  }
+  getAncestors(node: Node): Node[] {
+    const parentNode = this.hasParent(node) && this.getParent(node);
+    return parentNode ? [parentNode, ...this.getAncestors(parentNode)] : [];
+  }
+  getAttributeValue(elem: Element, name: string): string {
+    if (!this.isTag(elem)) {
+      return '';
+    }
+    const attrs = this.treeAdapter.getAttrList(elem);
+    for (const attr of attrs) {
+      if (attr.name === name) {
+        return attr.value;
+      }
+    }
+    return '';
+  }
+  getChildren(node: Node): Node[] {
+    return [...(this.treeAdapter.getChildNodes(node) || [])];
```

Comment:
Is the copy necessary? This is copying the empty array too.

Reply:
I want to be hygienic and not leak a reference to children array property.  However, the copy of the empty array was lazy.  I'm going with a version that doesn't copy now unless there's an array containing children to copy-- otherwise it returns a new empty array.

```ts
  getChildren(node: Node): Node[] {
    const children = this.treeAdapter.getChildNodes(node);
    return children && children.length > 0 ? [...children] : [];
  }
```